src/Display.js

```diff
@@ -49,29 +49,36 @@ class Display extends Component {
 
     constructor(props) {
         super()
-        const {itemWidth, itemHeight, total, buffer, paddingLeft, paddingTop, shouldPrerenderAll} = props
+        const {itemWidth, itemHeight, total, buffer, paddingLeft, paddingTop, shouldPrerenderAll, parentHeight, parentWidth} = props
         this.calculator = new GridCalculator({
             itemWidth,
             itemHeight,
             total,
             buffer,
             paddingLeft,
             paddingTop,
-            maxVisibleIndex: shouldPrerenderAll ? total : 0
+            maxVisibleIndex: shouldPrerenderAll ? total : 0,
+            displayHeight: parentHeight || 0,
+            displayWidth: parentWidth || 0
         })
 
         this.state = this.calculator.getState()
     }
 
     componentDidMount() {
+        const {parentHeight, parentWidth} = this.props
         this.updateDisplaySizeInState()
 
         this.display.addEventListener(`scroll`, createScrollListener(this))
-        window.addEventListener(`resize`, createWindowResizeListener(this))
+        if (!parentHeight || !parentWidth) {
```

Comment:
let's move this checking to separate method e.g `isParentSizeDefined(inst)` here and in `render` method